examples/speculative/speculative.cpp

```diff
@@ -64,6 +64,26 @@ int main(int argc, char ** argv) {
     params.n_gpu_layers = params.n_gpu_layers_draft;
     std::tie(model_dft, ctx_dft) = llama_init_from_gpt_params(params);
 
+    {
+        int n_vocab_tgt = llama_n_vocab(model_tgt);
+        if (n_vocab_tgt != llama_n_vocab(model_dft)) {
```

Comment:
This is not ideal. Codellama 7B and 13B have vocab size of 32000 while Codellama 34B has vocab size of 32016. It's the same vocab but with some extra tokens.

Reply:
How about we error if the size differs by more than 100, and also check the content of `min(n_vocab_tgt, n_vocab_dft)` tokens? Maybe even start 5 or something, to allow for cases where something like BOS or EOS has different content.

100 and 5 are just random numbers I plucked out of the air. The actual values can be whatever you prefer.

Reply:
Ok, thinks it would work

Reply:
I changed it. Also made the token content mismatch message a bit more helpful. For example, trying to use Orca 3B to draft Mistral 7B: 

    main: error: draft model vocab must match target model to use speculation but token 259 content differs - target '  ', draft ' t'